### PR TITLE
[EMB-13237] Allow to initialize report class without monitor installation

### PR DESCRIPTION
**Pull request by jdbandoni**

### CODE REVIEW

**Review by ArielDemarco — APPROVED** [review]
I'd just add some docs/comments so we don't forget what's the idea behind those changes

File: Source/KSCrash/Recording/KSCrashC.c [hunk 1]
```diff
@@ -481,3 +481,10 @@ void kscrash_deleteReportWithID(int64_t reportID)
 {
     kscrs_deleteReportWithID(reportID);
 }
+
+void kscrash_initializeReports(const char* appName, const char* const installPath) {
+    char path[KSFU_MAX_PATH_LENGTH];
+    snprintf(path, sizeof(path), "%s/Reports", installPath);
+    ksfu_makePath(path);
+    kscrs_initialize(appName, path);
```
Comment (blakdragan7) [inline Source/KSCrash/Recording/KSCrashC.c]: This changes internal global variables. Callling this more the once (i.e. for loading extension crashes from host app) will cause a problem.

File: Source/KSCrash/Recording/KSCrash.m [hunk 2]
```diff
@@ -141,24 +141,25 @@ + (instancetype) sharedInstance
     return sharedInstance;
 }
 
-+ (instancetype) sharedInstanceWithBasePath:(NSString *)basePath
++ (instancetype) sharedInstanceWithBasePath:(NSString *)basePath andBundleName:(NSString *)bundleName
```
Comment (blakdragan7) [inline Source/KSCrash/Recording/KSCrash.m:144]: If the goal of this change is to allow loading crashes from the extension app in the host app then this approach will not work. I recommend exposing a "static" function in KSCrash instance that directly access the internal "loading" of the crash report functions instead.

**Review by blakdragan7 — CHANGES_REQUESTED** [review]

**Comment by blakdragan7** [comment]
I do not think this is going to accomplish what you want it to. We can talk directly if you want to make a better way to do this.

**Comment by blakdragan7** [comment]
I just don't understand what the point of doing this is. Why do we need to set the bundle on the class instance ?

**Review by blakdragan7 — APPROVED** [review]

**Pull request merged**